src/models/simulation.ts

```diff
@@ -469,4 +500,29 @@ export class SimulationModel {
       this.sparks.length = 0;
     }
   }
+
+  @action.bound public updateCellsBaseStateFlag() {
+    this.cellsBaseStateFlag += 1;
+  }
+
+  @action.bound public updateCellsSimulationStateFlag() {
+    this.cellsSimulationStateFlag += 1;
+  }
+
+  public snapshot(): ISimulationSnapshot {
+    return {
+      time: this.time,
+      yearlyVegetationStatistics: this.yearlyVegetationStatistics,
```

Comment:
Okay, now I see you don't really copy the stats array. And also you never restore it. So this line should be removed. 

If you needed that for some reason, note that in all snapshot functions, you need to make sure you actually copy the object, not just reference. So, for an array you could say `yearlyVegetationStatistics: this.yearlyVegetationStatistics.slice()` or `yearlyVegetationStatistics: [...this.yearlyVegetationStatistics]`.